tests/host.c

```diff
@@ -48,6 +48,36 @@ void vector_inc(
     clReleaseContext(context);
 }
 
+void vector_inc_spirv(
```

Comment:
Let's try to duplicate less code. We can have a single `vector_inc` function, and use the test you have below (`bool is_spirv = (source_length >= 4) && (((uint32_t *)source)[0] == 0x07230203);`), to know which `clCreateProgram` function we should use.